npm-shrinkwrap.json

```diff
@@ -24,6 +24,17 @@
         "js-tokens": "^4.0.0"
       },
       "dependencies": {
+        "chalk": {
```

Comment:
@Page-, if I re-create the `npm-shrinkwrap.json` file through the following commands, I get a [different result](https://github.com/balena-io/balena-cli/pull/1650/commits/df8f7c212c42f333ea05e3ea2c4133b5a3d16f3e):
```
$ git checkout update-deps
$ git checkout master -- npm-shrinkwrap.json
$ rm -rf node_modules
$ npm install
```
```
$ node --version
v10.18.1
$ npm --version
6.13.4
```

My "different result" is in a draft PR (for the purpose of comparison / this discussion), based on the `update-deps` branch, which passes the build tests: https://github.com/balena-io/balena-cli/pull/1650
Skimming through the [differences](https://github.com/balena-io/balena-cli/pull/1650/commits/df8f7c212c42f333ea05e3ea2c4133b5a3d16f3e), I have the impression (could be wrong) that the shrinkwrap file in this PR updates/bumps more packages than the shrinkwrap file in the draft PR. Both satisfy the `package.json` requirements, but I think it could be said that the one in the draft PR is "more conservative", upgrading just as much as necessary to still satisfy the `package.json` file. Being conservative in this sense is probably a good thing: the "stability" gained from the shrinkwrap file.

Interestingly, `npm audit --production` appears to produce identical output in both cases.

What do you make of it? Out of interest, how did you create the `npm-shrinkwrap.json` file in this PR?

Reply:
@pdcastro I think what's happened with yours is that based on the starting point that npm decided to keep a load of packages duplicated, eg your PR adds an additional 4 copies of the `chalk` vs this PR, an extra 3 copies of `ansi-escapes`, one extra `strip-ansi`, 3 extra `supports-color`, and so on. If you try running `npm dedupe; npm i` on yours I'd be interested if it ended up more similar (the `npm i` would be to add the extraneous net-keepalive shrinkwrapped packages back).

As for how I created it I did it via a bunch of `npm update $package` and occasional `npm rm $package; npm i $package` to trigger clean installs with fully updated deps of those packages rather than keeping old versions due to them currently being installed - that being important in order to make sure we reduce the number of copies of modules to shrink the install size, and reduce the time spent loading and compiling duplicate modules (and reduce the associated memory usage)